File: Interviews/LeetCode/Topic-Wise/Greedy/Medium/Break-a-Palindrome/1328.py
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        ans, done, i, j = list(palindrome), False, 0, len(palindrome) - 1
        while i < j:
            if ans[i] != 'a':
                ans[i], done = 'a', True
                break
            i, j = i + 1, j - 1
        
        if not done:
            if len(palindrome) > 1: ans[-1] = 'b'
            else: ans = []
        
        return "".join(ans)

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-10/1328.cpp
class Solution {
public:
    string breakPalindrome(string palindrome) {
        bool done = false;
        string ans = palindrome;
        int i = 0, j = palindrome.size() - 1;
        while (i < j) {
            if (ans[i] != 'a') {
                ans[i] = 'a';
                done = true;
                break;
            }
            i += 1, j -= 1;
        }

        if (!done) {
            if (palindrome.size() > 1) {
                ans.back() = 'b';
            } else {
                ans = "";
            }
        }

        return ans;
    }
};